lib/helper.js

```diff
@@ -139,16 +139,35 @@ function withContext(Component, context) {
  * this method is not needed if you are using babel
  *
  * @method jsxTranspile
+ * @param {Boolean} instrument the original code
  **/
-function jsxTranspile() {
+function jsxTranspile(instrument) {
     var fs = require('fs');
     var babel = require('babel');
 
-    require.extensions['.jsx'] = function (module, filename) {
-        var content = fs.readFileSync(filename, 'utf8');
-        var compiled = babel.transform(content, {nonStandard: true});
-        return module._compile(compiled.code, filename);
-    };
+    if (instrument !== undefined && instrument && instrument === 'true') {
+        var istanbul = require('istanbul');
+        var coverageVariable = Object.keys(global).filter(
+            function (key) {
+                return /^\$\$cov_/.test(key);
+            })[0];
+        var instrumenter = new istanbul.Instrumenter({
+                coverageVariable: coverageVariable
+            });
+        require.extensions['.jsx'] = function (module, filename) {
+            var content = fs.readFileSync(filename, 'utf8');
+            var compiled = babel.transform(content, {nonStandard: true});
+            var instrumented = instrumenter.instrumentSync(compiled.code, filename);
```

Comment:
In my next PR, i will do inline for now. In the upcoming PR, we can read the dotfile, e.g. .jsx-test.yml.

ref implementation: https://github.com/nickmerwin/node-coveralls/blob/d5e1372bbd79a9fabbceec9624a220f1a8f75348/lib/getOptions.js#L72-L90